### PR TITLE
Allow null or undefined values in step result.

### PR DESCRIPTION
**Pull request by crism**
The following line could also be changed to `Array.isArray(value)`, but that is not necessary to fix the proximate problem.

### CODE REVIEW

**Comment by ryanjdew** [comment]
@crism The change looks good. I submitted a PR against your branch to add a unit test. Thanks for your contribution!

**Comment by crism** [comment]
PR accepted, @ryanjdew. Thanks!

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/suite-setup.xqy [hunk 1]
```diff
@@ -5,6 +5,14 @@ import module namespace test = "http://marklogic.com/test" at "/test/test-helper
 
 declare option xdmp:mapping "false";
 
+xdmp:invoke-function(function() {
+  xdmp:document-insert(
+    "/test/custom-null-step/main.sjs",
+    test:get-test-file("nullStep.sjs")
```
Comment (ryanjdew) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/suite-setup.xqy]:
```suggestion
    test:get-test-file("nullStep.sjs"),
    (xdmp:default-permissions(),xdmp:permission("flow-operator","execute"),xdmp:permission("flow-developer","execute")),
    ()
```

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/suite-setup.xqy [hunk 2]
```diff
@@ -5,6 +5,16 @@ import module namespace test = "http://marklogic.com/test" at "/test/test-helper
 
 declare option xdmp:mapping "false";
 
+xdmp:invoke-function(function() {
+  xdmp:document-insert(
+    "/test/custom-null-step/main.sjs",
+    test:get-test-file("nullStep.sjs"),
+    (xdmp:default-permissions(),xdmp:permission("flow-operator","execute"),xdmp:permission("flow-developer","execute")),
```
Comment (ryanjdew) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/suite-setup.xqy]:
```suggestion
    (xdmp:default-permissions(),xdmp:permission("flow-operator-role","execute"),xdmp:permission("flow-developer-role","execute")),
```

**Pull request merged**